modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java

```diff
@@ -700,7 +700,7 @@ public void onKernalStop() {
             /*force primary*/ !ctx.config().isReadFromBackup(),
             /*skip tx*/false,
             /*task name*/null,
-            /*deserialize binary*/false,
+            !ctx.keepBinary(),
```

Comment:
Why binary deserialization can be required here?

Reply:
To provide keys at exceptions at expected format.